### DISCUSSION TITLE
Create interactive SVG ...

### DISCUSSION THREAD

**Discussion by rleddy** [SVG-Edit/svgedit#596] [New feature request]
Looking for work or sponsorship as well...  [Sponsor copious-world](https://github.com/sponsors/copious-world?o=esb)

So, I have this concept of morphing SVG forms into interactive web pages (or for an electron app or similar).  I started this some time ago, and got something out of it. I was using InkScape.  [svg-forms](https://github.com/rleddy/svg-forms). But, it might be nice if something like Svelte could work with it - hence, a need for inserting directives into fairly parseable SVG.

There must be something out there that does not break the bank and works like [LiveCode](https://livecode.org/) but allows for lightweight web page interfaces with JavaScript and simple WASM insertion. But, I have not found it. Every time I sit down to do an interface I have to deal with a certain number of thoughts stemming from the axiomatic phrase "HERE WE GO AGAIN".

Most SVG tools seem geared to just drawing a picture.

**What I hope to do in the not too distant future**: I want to let people draw soft panels for their IoT devices. I started a thing **copious-blogs**. And, now I am using Svelte for the that. But, basically, the articles in a popup window can be SVG assemblies of things that connect to live machines. I have made progress. But, I need help, too. So, I have some interest in fixing up tools that speed it all along. But, also could use some delegation of work, also. Ultimately, composition and lazy loading should be as much a part of the processing as animation.

So, I am looking for work. Trying not to starve here.

I have done projects dealing with editors and interfaces before. I did this thing once with a product (a precursor of LiveCode) that imitated page editing programs (was it PageMaker? now long ago). It was for teaching. So, the whole PageMaker interface had to be done in this other system so that students could interact with editing tools and animations.  (I write in a variety of languages.)

Is there anybody associated with this project repository who can discuss possible projects to address any of these concepts?

**Comment by jfhenon** [comment]
Hello Richard, 
Actually, Optimistik started to "reactivate" SVGedit because we have an application (we call it OIMotion) that animates SVG based on external variables. For example, you can fill a tank based on a IOT sensor value, change a color if temperature is close to his high limit, etc...  We use it in dashboards for our industrial customers.
Your message is likely worth an offline discussion to understand how we could collaborate. Would you please leave a message on our website https://optimistik.io? 
Thanks
JF

**Reply by rleddy** [reply]
So, I will put the links to the Smashing article here: 
[animated svg](https://www.smashingmagazine.com/author/richard-leddy/)

I have done things like filling tanks. Started with some animation at the Kennedy Space Center.  One pagers are a thing of history now.

I want to bring components in from a search engine. I am halfway there with my current code. Hope to see some life in that fairly soon.

Today I just got my uploader storing stuff again for the blogs. 

For this space, besides introducing foreign objects or allowing some injection of component package attributes, I believe compositing is a key word. 

Here we are talking about human interaction.  In the past, I made an AI system that derived composites based on web searches to make movies. These were not much more than picture books with music in the background and some maps. But, the AI system made some sense. The guy I worked for got a patent for it.

If you have some idea for a future AI architecture, it is always possible to guide development in that direction from the beginning. End goals have a different flavor than say projects that are concerned about being ready for print.

**Comment by jfhenon** [comment]
just for information. We have a "demo/test" SVG that demonstrates various animation on a svg: 

https://user-images.githubusercontent.com/20402845/130238876-798c76da-58ee-413d-bc19-7809415a0ed8.mov
